azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -474,7 +474,8 @@ private boolean progressGraph() throws IOException {
     boolean jobsRun = false;
     for (final ExecutableNode node : nodesToCheck) {
       if (Status.isStatusFinished(node.getStatus())
```

Comment:
maybe consider putting all of the three conditions into a method called notReadyToRun() ?